### ISSUE TITLE
Random Comments not working

### ISSUE THREAD

**Issue by RohanKrishna500** [steevenakintilo/TwitterGiveawayBot#7]
So as the title says,i have pasted a particular term to search and it retweets and likes all the tweets containing it but it never comments under them.

**Comment by steevenakintilo** [comment]
Could you send me a screen of what the program print ?

**Comment by RohanKrishna500** [comment]
![image](https://github.com/steevenakintilo/TwitterGiveawayBot/assets/144703643/b716388f-a49a-4a71-bed4-cac570fb2b9c)

**Comment by RohanKrishna500** [comment]
> ![image](https://user-images.githubusercontent.com/144703643/268168573-b716388f-a49a-4a71-bed4-cac570fb2b9c.png)

recently new issues are generated related to account x,also now the bot is not doing any rt,likes or the specified comments.

**Comment by steevenakintilo** [comment]
> its on default ie 30 giveaways and crash and no is set to true
> 

Ok I have push modifications it should work

**Comment by steevenakintilo** [comment]
> > its on default ie 30 giveaways and crash and no is set to true
> 
> another issue arised something like Error searching for giveaways

The error is normal sometimes the twitter search will fail itself or won't find any giveway so if at the end the bot did do giveway it's fine

**Comment by RohanKrishna500** [comment]
> > > its on default ie 30 giveaways and crash and no is set to true
> > 
> > 
> > another issue arised something like Error searching for giveaways
> 
> The error is normal sometimes the twitter search will fail itself or won't find any giveway so if at the end the bot did do giveway it's fine

i have been trying it,but the error keeps occurring,i have given particular keyword to search so that it picks the giveaway that i am looking for,should i modify any thing in the configuration file?

**Comment by steevenakintilo** [comment]
> > > > its on default ie 30 giveaways and crash and no is set to true
> > > 
> > > 
> > > another issue arised something like Error searching for giveaways
> > 
> > The error is normal sometimes the twitter search will fail itself or won't find any giveway so if at the end the bot did do giveway it's fine
> 
> i have been trying it,but the error keeps occurring,i have given particular keyword to search so that it picks the giveaway that i am looking for,should i modify any thing in the configuration file?

Oh if the error keep occuring then set headless to False run the code again like for example 20 giveway and see what the browser do and after the program is done tell what happend and I will help you

**Comment by RohanKrishna500** [comment]
> > > > > its on default ie 30 giveaways and crash and no is set to true
> > > > 
> > > > 
> > > > another issue arised something like Error searching for giveaways
> > > 
> > > 
> > > The error is normal sometimes the twitter search will fail itself or won't find any giveway so if at the end the bot did do giveway it's fine
> > 
> > 
> > i have been trying it,but the error keeps occurring,i have given particular keyword to search so that it picks the giveaway that i am looking for,should i modify any thing in the configuration file?
> 
> Oh if the error keep occuring then set headless to False run the code again like for example 20 giveway and see what the browser do and after the program is done tell what happend and I will help you

changing headless to False gives me some error on the terminal,the thing what i am trying is to achieve is that the bot should comment  under a particular tweet which it will do by findind the keyword and also i have set some rt and likes parameter.But i just wanted to know that is there any way to spam comments under a tweet

**Comment by steevenakintilo** [comment]
> > > > > > its on default ie 30 giveaways and crash and no is set to true
> > > > > 
> > > > > 
> > > > > another issue arised something like Error searching for giveaways
> > > > 
> > > > 
> > > > The error is normal sometimes the twitter search will fail itself or won't find any giveway so if at the end the bot did do giveway it's fine
> > > 
> > > 
> > > i have been trying it,but the error keeps occurring,i have given particular keyword to search so that it picks the giveaway that i am looking for,should i modify any thing in the configuration file?
> > 
> > Oh if the error keep occuring then set headless to False run the code again like for example 20 giveway and see what the browser do and after the program is done tell what happend and I will help you
> 
> changing headless to False gives me some error on the terminal,the thing what i am trying is to achieve is that the bot should comment  under a particular tweet which it will do by findind the keyword and also i have set some rt and likes parameter.But i just wanted to know that is there any way to spam comments under a tweet

Oh I got it no the bot can only comment tweet linked to giveaway so you can't comment non related tweet

**Comment by RohanKrishna500** [comment]
well i dont think i quite get you,but its that tweet mentions to comment so i believe the bot should comment under it,i have actually commented already on that tweet but not using the bot,also is there any way to spam comments

**Comment by steevenakintilo** [comment]
Maybe but the bot can only do this:
- LookForGiveaway
- Like the tweet of the giveaway
- Rt the tweet of the giveaway
- Comment the tweet if it's needed
- Follow account that we need to follow to enter giveaway
- Then do random retweet and tweet at the end to act more "human"
The bot can't spam comments and even it could it could lead your account to a ban for spaming so it's not a good solution

**Comment by RohanKrishna500** [comment]
ok i get it,can the bot do giveaways of a particular account,like it can avoid the ones it has already done of the same account but do other giveaways of that account by having entered the keyword

**Comment by RohanKrishna500** [comment]
> Maybe but the bot can only do this:
> 
> * LookForGiveaway
> * Like the tweet of the giveaway
> * Rt the tweet of the giveaway
> * Comment the tweet if it's needed
> * Follow account that we need to follow to enter giveaway
> * Then do random retweet and tweet at the end to act more "human"
>   The bot can't spam comments and even it could it could lead your account to a ban for spaming so it's not a good solution

Comment the tweet if it's needed,this part is where it messes up,like for some giveaway it did comment but that too only once,but for a particular giveaway it doesnt comment at all,even though in config file i changed the word to look for commenting

**Comment by steevenakintilo** [comment]
oh I see thanks could you tell me what comment word you put for the word to comment word_list_to_check_for_comment:
 on the config file and could you send me the link of the tweet please ?

**Comment by RohanKrishna500** [comment]
i put "Comment" as  a word in the word_list_to_check_for_comment
tweet link - https://twitter.com/AmreliaRuhez/status/1700381160587821469

**Comment by RohanKrishna500** [comment]
hello,i just wanted to know that can the bot run on vps and what exactly does forever_loop mean?

**Comment by steevenakintilo** [comment]
If you can install python and chrome then yes and forever_loop if you set it to true the program will never stop just after doing giveaway doing a sleep between 1 and 2 days and run again etc etc

**Comment by RohanKrishna500** [comment]
ok i got a couple of questions on that -
1)does that mean i need to my machine to run always
2)if i search for 30 giveaways and after all those 30 giveaways are finished,will it refresh and search and do some more giveaways based on the same search term
3)install python ok,chrome means like should i put headless as true?

**Comment by steevenakintilo** [comment]
Sorry for slow answer forgot to answer you

1) No you don't need to run your machine always just run your machine when you run the script
2) If you search for 30 giveaways it will make them then stop the program
3) If you put headless to true the code won't display a chrome windows of what it's doing so you should put it to true

**Comment by RohanKrishna500** [comment]
well in my case once i close the console the script stops working and it does not run on a forever loop so i thought of running the script on virtual private server.Hope u get what I am saying

**Comment by steevenakintilo** [comment]
Oh yes if you close the console it would close the bot it's normal and if you want to run it forever buying a private server to host It is a good idea

**Comment by RohanKrishna500** [comment]
Hi,I got this bug while using the bot with multiple accounts
Starting Twitter
reloging failed
wrong username of password
skipping the account
End of the program

i tried to to retype the same username again and it got fixed even though its the same as it was earlier
Occurred twice today
Any Fix?

**Comment by steevenakintilo** [comment]
Thanks for the info I will fix this part 
You mean only using one account to tag multiple account ?

**Comment by RohanKrishna500** [comment]
> You mean only using one account to tag multiple account ?
I mean as i am using multiple twitter accounts in the config and under the accounts_to_tag,is it possible each of my twitter accounts be able to tag unique accounts and not the same accounts as all the other accounts
For Ex:-
#Account info write the username of all your account
account_username:
  - "test1234"
  - "test4444"
  - "test0000"

#Account info write the password of all your account
account_password:
  - "twitter1234"
  - "twitter4444"
  - "twitter0000"

these are the multiple accounts i am using and for each of the individual account it should tag unique accounts and not share the same among themselves

**Comment by steevenakintilo** [comment]
Oh I understand its a nice feature I should add but how would I know which account to tag is for which account ?

**Comment by RohanKrishna500** [comment]
even i dont know how it will work,i guess it you could use the same  way you mapped the twitter account to the password.

**Comment by RohanKrishna500** [comment]
Just got an another bug where the bot tags @twitter account even though its not in config file and avoid the account to be tagged which is mentioned in config file

**Comment by steevenakintilo** [comment]
Oh thats weird could you send screenshot

**Comment by RohanKrishna500** [comment]
well it does not show any error or something in the terminal so i didnt take screenshot.

**Comment by steevenakintilo** [comment]
I meant not on the terminal but on your twitter account

**Comment by RohanKrishna500** [comment]
next time i get the error i will surely screenshot it,as of now I didnt get the tweet after searching for ir

**Comment by steevenakintilo** [comment]
ok ok no problem mate

**Comment by RohanKrishna500** [comment]
Hi,got this issue recently where the bot says giveaways not found or error searching for a tweet even though there is a giveaway.
![image](https://github.com/steevenakintilo/TwitterGiveawayBot/assets/144703643/8b8303bb-0c96-49b4-a079-87b993c2b319)

**Comment by steevenakintilo** [comment]
oh what were the search word you were using ?

**Comment by RohanKrishna500** [comment]
"kiwi melody"
and likes and retweet to 0
and giveaway word retrival tried between 0 to 10 all combo
and max giveaway from 0 to 10 all combo

**Comment by steevenakintilo** [comment]
ok ok I will try myself and see why it doesn't work when I have time

**Comment by steevenakintilo** [comment]
I've already tried to fix this problem but it seems to be a random issue even myself I barely got this error in months of running

**Comment by RohanKrishna500** [comment]
one thing i noticed is that if i tweet contains emoji's or any non text based info,it dosent consider it,also if i want to search for a word that word is to be present as the first word in the tweet itself(like if its in the middle of two words it cannot retrieve it)

**Comment by steevenakintilo** [comment]
Yes I have removed emoji from comment/tweet it was giving me errors, which type of tweet you want to search? Like give me example

**Comment by RohanKrishna500** [comment]
the same i mention above "kiwi melody",that tweet contains this word

**Comment by steevenakintilo** [comment]
I could try adding this features in the future If I have time for it

**Comment by RohanKrishna500** [comment]
i got an error today
![image](https://github.com/steevenakintilo/TwitterGiveawayBot/assets/144703643/767eb188-6d82-41a5-b974-a8fe5ac61a96)

**Comment by steevenakintilo** [comment]
Sometimes those type of error can happen but it's not a big deal

**Comment by RohanKrishna500** [comment]
> Sometimes those type of error can happen but it's not a big deal

i can bear with these,just wanted to tell you

**Comment by RohanKrishna500** [comment]
hi,can this bot send img files and tag accounts together in a single tweet?

**Comment by steevenakintilo** [comment]
The bot can't send img files sadly

**Comment by RohanKrishna500** [comment]
hi,so after a long time i tried the bot and now i get an error which wasnt there previously
D:\Twitter Bot\advanced twitter bot 1>    pip install -r requirements.txt
Defaulting to user installation because normal site-packages is not writeable
Collecting emoji==2.2.0 (from -r requirements.txt (line 1))
  Using cached emoji-2.2.0.tar.gz (240 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Collecting feedparser==6.0.10 (from -r requirements.txt (line 2))
  Using cached feedparser-6.0.10-py3-none-any.whl.metadata (2.3 kB)
Collecting PyYAML==6.0 (from -r requirements.txt (line 3))
  Using cached PyYAML-6.0.tar.gz (124 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... error
  error: subprocess-exited-with-error

  × Getting requirements to build wheel did not run successfully.
  │ exit code: 1
  ╰─> [54 lines of output]
      running egg_info
      writing lib\PyYAML.egg-info\PKG-INFO
      writing dependency_links to lib\PyYAML.egg-info\dependency_links.txt
      writing top-level names to lib\PyYAML.egg-info\top_level.txt
      Traceback (most recent call last):
        File "C:\Python312\Lib\site-packages\pip\_vendor\pyproject_hooks\_in_process\_in_process.py", line 353, in <module>
          main()
        File "C:\Python312\Lib\site-packages\pip\_vendor\pyproject_hooks\_in_process\_in_process.py", line 335, in main
          json_out['return_val'] = hook(**hook_input['kwargs'])
                                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "C:\Python312\Lib\site-packages\pip\_vendor\pyproject_hooks\_in_process\_in_process.py", line 118, in get_requires_for_build_wheel
          return hook(config_settings)
                 ^^^^^^^^^^^^^^^^^^^^^
        File "C:\Users\rohan\AppData\Local\Temp\pip-build-env-dmay9573\overlay\Lib\site-packages\setuptools\build_meta.py", line 334, in get_requires_for_build_wheel
          return self._get_build_requires(config_settings, requirements=[])
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "C:\Users\rohan\AppData\Local\Temp\pip-build-env-dmay9573\overlay\Lib\site-packages\setuptools\build_meta.py", line 304, in _get_build_requires
          self.run_setup()
        File "C:\Users\rohan\AppData\Local\Temp\pip-build-env-dmay9573\overlay\Lib\site-packages\setuptools\build_meta.py", line 320, in run_setup
          exec(code, locals())
        File "<string>", line 288, in <module>
        File "C:\Users\rohan\AppData\Local\Temp\pip-build-env-dmay9573\overlay\Lib\site-packages\setuptools\__init__.py", line 117, in setup
          return distutils.core.setup(**attrs)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "C:\Users\rohan\AppData\Local\Temp\pip-build-env-dmay9573\overlay\Lib\site-packages\setuptools\_distutils\core.py", line 183, in setup
          return run_commands(dist)
                 ^^^^^^^^^^^^^^^^^^
        File "C:\Users\rohan\AppData\Local\Temp\pip-build-env-dmay9573\overlay\Lib\site-packages\setuptools\_distutils\core.py", line 199, in run_commands
          dist.run_commands()
        File "C:\Users\rohan\AppData\Local\Temp\pip-build-env-dmay9573\overlay\Lib\site-packages\setuptools\_distutils\dist.py", line 954, in run_commands
          self.run_command(cmd)
        File "C:\Users\rohan\AppData\Local\Temp\pip-build-env-dmay9573\overlay\Lib\site-packages\setuptools\dist.py", line 995, in run_command
          super().run_command(command)
        File "C:\Users\rohan\AppData\Local\Temp\pip-build-env-dmay9573\overlay\Lib\site-packages\setuptools\_distutils\dist.py", line 973, in run_command
          cmd_obj.run()
        File "C:\Users\rohan\AppData\Local\Temp\pip-build-env-dmay9573\overlay\Lib\site-packages\setuptools\command\egg_info.py", line 313, in run
          self.find_sources()
        File "C:\Users\rohan\AppData\Local\Temp\pip-build-env-dmay9573\overlay\Lib\site-packages\setuptools\command\egg_info.py", line 321, in find_sources
          mm.run()
        File "C:\Users\rohan\AppData\Local\Temp\pip-build-env-dmay9573\overlay\Lib\site-packages\setuptools\command\egg_info.py", line 544, in run
          self.add_defaults()
        File "C:\Users\rohan\AppData\Local\Temp\pip-build-env-dmay9573\overlay\Lib\site-packages\setuptools\command\egg_info.py", line 582, in add_defaults
          sdist.add_defaults(self)
        File "C:\Users\rohan\AppData\Local\Temp\pip-build-env-dmay9573\overlay\Lib\site-packages\setuptools\command\sdist.py", line 109, in add_defaults
          super().add_defaults()
        File "C:\Users\rohan\AppData\Local\Temp\pip-build-env-dmay9573\overlay\Lib\site-packages\setuptools\_distutils\command\sdist.py", line 238, in add_defaults
          self._add_defaults_ext()
        File "C:\Users\rohan\AppData\Local\Temp\pip-build-env-dmay9573\overlay\Lib\site-packages\setuptools\_distutils\command\sdist.py", line 323, in _add_defaults_ext
          self.filelist.extend(build_ext.get_source_files())
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "<string>", line 204, in get_source_files
        File "C:\Users\rohan\AppData\Local\Temp\pip-build-env-dmay9573\overlay\Lib\site-packages\setuptools\_distutils\cmd.py", line 107, in __getattr__
          raise AttributeError(attr)
      AttributeError: cython_sources
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
error: subprocess-exited-with-error

× Getting requirements to build wheel did not run successfully.
│ exit code: 1
╰─> See above for output.

note: This error originates from a subprocess, and is likely not a problem with pip.

D:\Twitter Bot\advanced twitter bot 1>
what to do for this

**Comment by steevenakintilo** [comment]
Hello try this 
![image](https://github.com/user-attachments/assets/71776469-ad50-41f0-aaa2-230c6ba52870)
![image](https://github.com/user-attachments/assets/4c159ef1-f672-4346-906b-47929d07bc2c)
![image](https://github.com/user-attachments/assets/30dbd779-c981-433a-b510-c2ffbb0b4a4c)

**Comment by RohanKrishna500** [comment]
i isntalled python version 3.10 and its working but i got an another issue where it skips account even if i have put the account in splits,
![image](https://github.com/user-attachments/assets/5ad88b41-8316-41b5-8ba8-6aa668bc3de2)
credentials are all correct still i get this

**Comment by steevenakintilo** [comment]
does it happend every time?

**Comment by RohanKrishna500** [comment]
yes it not working for me for a long time.i havent changed password or anything to any of these accounts

**Comment by steevenakintilo** [comment]
I've run the bot yesterday was working fine.
Try create a new account and test it out and tell me if it's still fail.

**Comment by RohanKrishna500** [comment]
alr

**Comment by RohanKrishna500** [comment]
hi i tried another account of mine and it still failed,is ig twitter just blocked my ip or something so why login in with bots dont work

**Comment by steevenakintilo** [comment]
Yes probably because I use my bot daily and never got this error should be an account problem
